### DISCUSSION TITLE
1.5.2 will not compile a sketch on Arduino 2.0.3

### DISCUSSION THREAD

**Discussion by michaellingg** [SpenceKonde/DxCore#385] [General]
Greetings,

I get the below error messages when trying to compile a sketch that worked prior to the latest library update.

Nothing compiles, not even the canned examples.

Any thoughts and or help would be greatly appreciated!

-michael

>>>
HW: Microchip AVR-IoT-Cellular mini

SW: Arduino IDE 2.0.3
DxCore 1.5.2
Lib: Microchip AVR-IoT-Cellular V1.3.3

C:\Users\michaell\AppData\Local\Arduino15\packages\DxCore\hardware\megaavr\1.5.2\libraries\DxCore\src/DxCore.h:66:20: error: variable or field 'configXOSC32K' declared void
 void configXOSC32K(X32K_TYPE_t settings = X32K_HIGHPWR_START2S, X32K_ENABLE_t enable = X32K_ENABLED);
   
C:\Users\michaell\AppData\Local\Arduino15\packages\DxCore\hardware\megaavr\1.5.2\libraries\DxCore\src/DxCore.h:66:20: error: 'X32K_TYPE_t' was not declared in this scope
C:\Users\michaell\AppData\Local\Arduino15\packages\DxCore\hardware\megaavr\1.5.2\libraries\DxCore\src/DxCore.h:66:20: note: suggested alternative: 'X32K_OPT_t'
 void configXOSC32K(X32K_TYPE_t settings = X32K_HIGHPWR_START2S, X32K_ENABLE_t enable = X32K_ENABLED);

C:\Users\michaell\AppData\Local\Arduino15\packages\DxCore\hardware\megaavr\1.5.2\libraries\DxCore\src/DxCore.h:66:79: error: expected primary-expression before 'enable'
 void configXOSC32K(X32K_TYPE_t settings = X32K_HIGHPWR_START2S, X32K_ENABLE_t enable = X32K_ENABLED);
<<<

**Comment by SpenceKonde** [comment]
Which canned examples do not work?

What the hell hardware is that? How does it relate to my core???? This core does not provide product-level hardware definitions, we provide chip level ones. I've never heard of that board, much less know how it has anything to do with DxCore. 

2.0.3 of the IDE is not guaranteed to work, no work has been conducted to add such support, and due to the critical priorities I list below, all of which have higher priority, I do not expect to have time to make 2.x work possibly until the start of the second quarter. 

In any case - unless the problem is demonstrated to meet these criteria:
* Occur in 1.8.13 (this is our reference "good release" - there has been no release since without regressions which added nothing in terms of functionality that is visible to me, but caused multiple regressions that were. Ergo, 1.8.13 is the only recommended IDE version at this time. 
* Not occur in the final 1.4.x version of DxCore
* Cause issues outside of corner cases which severely impact usability many users.
* Has a test case known to fail made available to me

Only issues that meet all three of those criteria can be triaged and might take priorirty over the following list of critical projects.

The current list of things that must be done before I look into anything but the most severe 
 * 1.5.3 DxCore critical release to keep burn bootloader from bricking AVR-DD series parts, to be released shortly. 
 * 2.6.5 mTC release to pull in some neatly completed work.
 * Ensure that ATTinyCore 2.0.0 is released (sorry, they have been waiting too long for a release, there are more users there and they have gotten no release in over a year. Their most recent release has severe deficiencies. 

Thereafter, there is a list of critical projects which must be completed prior to further work on the modern AVR cores, they are:
* Set up patreon and github sponsors fully to begin bringing in revenue with which to deal with that guy who claims to be the lord of the land and demands money every month. 
* Completing listings for tindie for hardware that is ready to sell but for lack of description. 
* Diagnose and fix mission critical on-site server. 
* Completing the remaining tindie listings; these are identical to the above case, except that they also need product photograph, the mission critical hardware without which this process is untenable must be diagnosed and repaired, as it plays a central role in all operations. Photography is one of the many things which cannot be realized until this hardware issue has been corrected. 

Only one ALL of those CRITICAL AND URGENT matters has been corrected can I look at supporting a new major version of an IDE from an organization which had only a single release version of 1.8.x that was fit for purpose in the past 2 years. Arduino is constantly breaking things, and you should assume that, unless there is a broad consensus that the version works both in general and with the hardware package you are using, a new IDE release should be assumed to not work correctly - the Arduino trackrecord is just so poor in terms of release quality that like, that I must advise against upgrading until there is a clear consensus that the release is fit for purpose. Their hit rate is around 10-15% (of their releases that are suitable for use), and I would think it would be much lower still when

**Reply by michaellingg** [reply]
Spence,

Thanks for the guidance!  I could not agree more with you about Arduino IDE in general.  I have found 1.8.13 and will give it a go tonight.

Microchip who produces the board i am using directly references DxCore.  Here is the board, seems to be a nice bit of kit.
https://www.microchip.com/en-us/solutions/internet-of-things/cellular-iot

Not working examples: debug_moden in the AVR-IoT-Cellular pull down.  Since i have already written several modules that used to work i won't bother you with those.

I did not intend to pile more on your plate!  Thanks for the work (DxCore) and the guidance!

-m

**Reply by pcfreak1201** [reply]
Hi Michael,

I saw the board in the Adafruit video and it really looks interesting. I followed the steps from microchip:(https://iot.microchip.com/docs/arduino/introduction/devenv)
and on IDE 1.8.15 it compiles:
```
Bibliothek AVR-IoT-Cellular in Version 1.3.3 im Ordner: /home/schulte/Arduino/libraries/AVR-IoT-Cellular  wird verwendet
Bibliothek Wire in Version 2.0.9 im Ordner: /home/schulte/.arduino15/packages/DxCore/hardware/megaavr/1.5.2/libraries/Wire  wird verwendet
/home/schulte/.arduino15/packages/DxCore/tools/avr-gcc/7.3.0-atmel3.6.1-azduino6/bin/avr-size -A /tmp/arduino_build_775899/debug_modem.ino.elf
Der Sketch verwendet 7608 Bytes (5%) des Programmspeicherplatzes. Das Maximum sind 131072 Bytes.
Globale Variablen verwenden 3630 Bytes (22%) des dynamischen Speichers, 12754 Bytes für lokale Variablen verbleiben. Das Maximum sind 16384 Bytes.
```

**Reply by ITDiscovery** [reply]
On the pictures I am seeing posted by Microchip, it's an AVR128DB48....looks like it has a 25CSM04 (a 4MB EEProm) and the standard debugger along with the Cellphone modem. I'd be curious if you tried both Arduino IDEs using the AVR128DB48....not sure how you'd load the bootloader via that debugger...

Looks cool...I just got a 23LCS512 going via SPI and bit-bang, so that looks like a really good marriage. $62 from Newark in 2023 money. You gotta get cellular service though.. :-(

**Comment by M65649** [comment]
Hello,

I'm one of the main maintainers on the AVR-IoT Cellular library. Please feel free to ask any questions or open up issues on our Github page. We're happy to help with any issues.

Regarding the compile issues. I'm not able to reproduce it with Arduino IDE 2.0.3, DxCore 1.5.3 and AVR-IoT Cellular Library 1.3.4, but as said, please feel free to contact us if there are any issues.
